### PR TITLE
Conditionally pack symbols based on keepNativeSymbols

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/runtime/issues/58455#issuecomment-911665744

Source build runs with `keepNativeSymbols=true` which prevents the symbols from being stripped out and placed into a separate .dbg files.  This supports the normal workflows used by distro package maintainers.  You can read more background [here](https://github.com/dotnet/source-build/pull/1776/files#diff-3cc321025ad7a47e73d552f56eba06e9fb93de57daff50548b9c5e3a98febdad).  This change conditions `TargetsForTfmSpecificDebugSymbolsInPackage` on the `keepNativeSymbols`.

FYI, I will port this to main as well once this PR gets approval.

### CODE REVIEW

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <DebugType>none</DebugType>
     <IsPackable>true</IsPackable>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>$(SymbolsSuffix)</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
-    <TargetsForTfmSpecificDebugSymbolsInPackage>$(TargetsForTfmSpecificDebugSymbolsInPackage);AddRuntimeSpecificNativeSymbolToPackage</TargetsForTfmSpecificDebugSymbolsInPackage>
+    <TargetsForTfmSpecificDebugSymbolsInPackage Condition="'$(keepNativeSymbols)' != 'true'">$(TargetsForTfmSpecificDebugSymbolsInPackage);AddRuntimeSpecificNativeSymbolToPackage</TargetsForTfmSpecificDebugSymbolsInPackage>
```
Comment (ViktorHofer) [inline src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:11]: Is `keepNativeSymbols` an environment variable? Who sets it? Asking as it looks very strange for an msbuild property.

Comment (MichaelSimons) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:11]: It gets set by source build [here](https://github.com/dotnet/runtime/blob/release/6.0/eng/SourceBuild.props#L47)

```
<InnerBuildArgs>$(InnerBuildArgs) /p:KeepNativeSymbols=true</InnerBuildArgs>
```

Comment (ViktorHofer) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:11]: Cool. Should we fix the casing then? :)

```suggestion
    <TargetsForTfmSpecificDebugSymbolsInPackage Condition="'$(KeepNativeSymbols)' != 'true'">$(TargetsForTfmSpecificDebugSymbolsInPackage);AddRuntimeSpecificNativeSymbolToPackage</TargetsForTfmSpecificDebugSymbolsInPackage>
```

Comment (ViktorHofer) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:11]: Maybe also add a comment here explaining what this does?

Comment (MichaelSimons) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props]: Fixed casing and added comment.

**Review by ViktorHofer — APPROVED** [review]
Awesome, thanks Michael.

**Comment by ViktorHofer** [comment]
cc @danmoseley, tell-mode infra change for source-build.

**Pull request merged**